lib/matplotlib/scale.py

```diff
@@ -82,6 +82,88 @@ def get_transform(self):
         return IdentityTransform()
 
 
+class ArbitraryTransform(Transform):
+     """
+     A simple transform that takes and arbitrary function for the
+     forward and inverse transform.
+     """
+
+     input_dims = 1
+     output_dims = 1
+     is_separable = True
+     has_inverse = True
+
+     def __init__(self, forward, inverse):
+         """
+         Parameters
+         ----------
+
+         forward: The forward function for the transform
+
+         inverse: The inverse of the forward function.
+         """
+         super().__init__()
+         if callable(forward) and callable(inverse):
+             self._forward = forward
+             self._inverse = inverse
+         else:
+             raise ValueError('arguments to ArbitraryTransform must '
+                              'be functions.')
+
+     def transform_non_affine(self, values):
+         with np.errstate(divide='ignore', invalid='ignore'):
+             q = self._forward(values)
+         return q
+
+     def inverted(self):
+         return ArbitraryTransform(self._inverse, self._forward)
+
+
+class ArbitraryScale(ScaleBase):
+
+     name = 'arbitrary'
+
+     def __init__(self, axis, functions=None):
+         """
+         Provide an arbitrary scale for the axis.
+
+         Parameters
+         ----------
+
+         axis: the axis for the scale
+
+         functions: (forward, inverse)
+            two-tuple of the forward and inverse functions for the scale.
+
+        Examples
+        --------
+            TODO
+
+         """
+         if functions is None or len(functions) < 2:
```

Comment:
I'd probably just write `forward, inverse = functions` and get whatever error message you have when functions is not an iterable of len 2, rather than supplying your own error message.  (After all, as of this commit passing a non-interable will also give you the builtin message.)